lib/system/system.toit

```diff
@@ -247,3 +253,28 @@ If the program is run as an executable, this is the fully resolved path to the
 */
 program-path -> string?:
   #primitive.core.program-path
+
+/**
+The hostname of the machine running the program.
+*/
+hostname -> string:
+  if platform == PLATFORM-FREERTOS:
+    config-name := bucket_.get "hostname"
+    if config-name: return config-name
+  return hostname_
+
+/**
+Sets the hostname of the machine running the program.
+
+This operation is not supported on all platforms.
+
+Only new network connections will use the new hostname. Also, some
+  routers may cache the old hostname for a while.
+*/
+hostname= hostname/string -> none:
+  if platform != PLATFORM-FREERTOS:
+    throw "Setting hostname is not supported on this platform"
```

Comment:
We typically use much shorter exceptions in the core libraries. 

```suggestion
    throw "UNSUPPORTED"
```

Reply:
done.